### DISCUSSION TITLE
Tidalapi appears abandoned.

### DISCUSSION THREAD

**Discussion by tehkillerbee** [EbbLabs/mopidy-tidal#107] [Ideas]
Hi guys, 

Tidalapi appears abandoned. Therefore, I would like to hear your opinions on how the users and contributors of mopidy-tidal. 

One suggestion could be to create a fork of tidalapi and create a new PyPi package for it. But I would rather try to encourage Tamland to add more maintainers to the project. 

Any good ideas @2e0byo @BlackLight?

I understand everyone's time is precious so maintaining yet another open source project is not ideal and I do not expect anyone to take such task. :)

**Comment by blacklight** [comment]
I'm already a bit familiar with python-tidal's codebase since I've already had to tweak it several times.

I can definitely help as a sporadic contributor - like if something breaks, as it recently did, or some major new feature requires upstream changes, I'm happy to put together a PR.

I also contribute to or maintain a lot projects already though, so anything that entails more work/responsibilities than the "sporadic contributor" status may be a bit of problem for me in terms of available bandwidth.

The important thing IMHO is that, if something breaks on python-tidal (again, as it recently did), me or somebody else should be allowed to push a PR and get it approved fast without waiting weeks/months for Tamland's approval. The current situation is even less ideal for me: it already happened twice that things broke, I pushed PRs that took too long to be approved, eventually more and more people start using my fork, and that puts me in the slightly more uncomfortable position of "involuntary temporary maintainer".

I also agree that forking the project for good may sound like a tempting option sometimes, but one that I would avoid on the long term: once it's officially forked we're on our own.

**Comment by 2e0byo** [comment]
I use mopidy-tidal every day and am happy to contribute to keeping it alive.  I'd offer to maintain a fork if we went that way *but* I'm finishing a PhD in a month's time and don't know what my life will look like later, so I should probably come back in a month and have a look.  On the other hand I don't currently maintain anything of a significant size.

If it were a matter of being *added* as a maintainer I'd be happy.  Tidalapi has a decent testsuite, but it doesn't currently run in CI.  Just doing that would make handling PRs easier.

I agree with @BlackLight  about the problem with forking.  But the thing with github's (and most forge's) fork-and-pr structure is that we *have* forked and end up de facto maintaining.  I don't see a good solution to that.

Tidalapi is pretty clean already.   We could move the code we use into mopidy-tidal and drop the rest, but I don't think we'd gain much.

Do either of you have a way to reach out to Tamland/Modguldir or should we open an issue?

**Comment by tehkillerbee** [comment]
> The important thing IMHO is that, if something breaks on python-tidal (again, as it recently did), me or somebody else should be allowed to push a PR and get it approved fast without waiting weeks/months for Tamland's approval. The current situation is even less ideal for me: it already happened twice that things broke, I pushed PRs that took too long to be approved, eventually more and more people start using my fork, and that puts me in the slightly more uncomfortable position of "involuntary temporary maintainer".

Very good point. When we switch temporarily to your fork, you end up becoming a maintainer until Tamland comes back which he has done a few times. I've been against this from the beginning - and have yet to switch in the last mopidy-tidal release to avoid pushing everyone over on your fork.

> I also agree that forking the project for good may sound like a tempting option sometimes, but one that I would avoid on the long term: once it's officially forked we're on our own.

Also true. But at the moment I am not certain if tidalapi has been abandoned, it has been a while since I wrote Tamland and he has yet to react on the issue.

@2e0byo I agree, one part is maintaining it but another is taking over the project altogether. I had to do it with mopidy-tidal, when it was abandoned, but that has of course been a much smaller task. Also since I have not contributed that much code myself - only maintaining and reviewing PRs

> Do either of you have a way to reach out to Tamland/Modguldir or should we open an issue?

No, unfortunately. I have already created an issue a while back. It does not look like he has reacted. 

EDIT:  In any case, I should probably clarify that IF a fork is necessary, I do expect to be able to help contributing/maintaining the project, in a similar fashion as I have tried to do with mopidy-tidal. So I do not expect anyone to take over the task of being the sole maintainer.
